molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java

```diff
@@ -463,10 +463,10 @@ private void upsertAttributes(EntityType entityType, EntityType existingEntityTy
     // analyze both compound and atomic attributes owned by the entity
     Map<String, Attribute> attrsMap =
         stream(entityType.getOwnAllAttributes())
-            .collect(toMap(Attribute::getName, Function.identity()));
+            .collect(toMap(Attribute::getIdentifier, Function.identity()));
     Map<String, Attribute> existingAttrsMap =
         stream(existingEntityType.getOwnAllAttributes())
-            .collect(toMap(Attribute::getName, Function.identity()));
+            .collect(toMap(Attribute::getIdentifier, Function.identity()));
 
     // determine attributes to add, update and delete
     Set<String> addedAttrNames = Sets.difference(attrsMap.keySet(), existingAttrsMap.keySet());
```

Comment:
Also update the names of these variables now the maps don't store names anymore.